File: backend/models/order.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// Order 表示用户的订单
// 包含订单状态、支付信息和总金额
type Order struct {
	ID        uint           `gorm:"primarykey" json:"id"` // Override ID to ensure lowercase JSON
	CreatedAt time.Time      `json:"created_at"`           // Override for custom formatting if needed, but here just for key name
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`

	OrderNo     string      `gorm:"uniqueIndex;not null" json:"order_no"` // 订单编号，唯一
	UserID      uint        `json:"user_id"`                              // 关联的用户 ID
	TotalAmount float64     `json:"total_amount"`                         // 订单总金额
	Status      int         `gorm:"default:0" json:"status"`              // 订单状态: 0-待支付, 1-待发货, 2-待收货, 3-待评价(已完成), 4-已取消, 5-售后中
	AddressID   uint        `json:"address_id"`                           // 收货地址 ID
	Address     Address     `json:"address"`                              // 收货地址快照 (简化处理，实际应复制地址信息)
	Items       []OrderItem `gorm:"foreignKey:OrderID" json:"items"`      // 订单包含的商品项
}

// OrderItem 表示订单中的具体商品项
type OrderItem struct {
	ID        uint           `gorm:"primarykey" json:"id"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`

	OrderID      uint    `json:"order_id"`      // 关联的订单 ID
	ProductID    uint    `json:"product_id"`    // 商品 ID
	ProductName  string  `json:"product_name"`  // 商品名称 (快照)
	ProductImage string  `json:"product_image"` // 商品图片 (快照)
	SKUID        uint    `json:"sku_id"`        // SKU ID
	SKUName      string  `json:"sku_name"`      // SKU 名称 (快照)
	Price        float64 `json:"price"`         // 购买时的单价
	Quantity     int     `json:"quantity"`      // 购买数量
}

File: app/lib/features/order/models/order.dart
/// 订单模型
class Order {
  final int id;
  final String orderNo;
  final double totalAmount;
  final int status; // 0: Pending, 1: Paid, etc.
  final String createdAt;
  final List<OrderItem> items;

  Order({
    required this.id,
    required this.orderNo,
    required this.totalAmount,
    required this.status,
    required this.createdAt,
    required this.items,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['id'],
      orderNo: json['order_no'],
      totalAmount: (json['total_amount'] as num).toDouble(),
      status: json['status'],
      createdAt: json['created_at'],
      items: (json['items'] as List)
          .map((item) => OrderItem.fromJson(item))
          .toList(),
    );
  }

  // 获取状态描述
  String get statusText {
    switch (status) {
      case 0:
        return '待付款';
      case 1:
        return '待发货';
      case 2:
        return '待收货';
      case 3:
        return '待评价';
      case 4:
        return '已完成';
      case 5:
        return '售后中';
      case -1:
        return '已取消';
      default:
        return '未知状态';
    }
  }
}

/// 订单项模型
class OrderItem {
  final int id;
  final String productName;
  final double price;
  final int quantity;
  final String productImage;

  OrderItem({
    required this.id,
    required this.productName,
    required this.price,
    required this.quantity,
    required this.productImage,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) {
    return OrderItem(
      id: json['id'],
      productName: json['product_name'],
      price: (json['price'] as num).toDouble(),
      quantity: json['quantity'],
      productImage: json['product_image'],
    );
  }
}
